pkg/kms/localkms/localkms_writer.go

```diff
@@ -17,21 +18,82 @@ import (
 const maxKeyIDLen = 20
 
 // newWriter creates a new instance of local storage key storeWriter in the given store and for masterKeyURI
-func newWriter(kmsStore storage.Store) *storeWriter {
+func newWriter(kmsStore storage.Store, opts ...PrivateKeyOpts) *storeWriter {
+	pOpts := &privateKeyOpts{}
+
+	for _, opt := range opts {
+		opt(pOpts)
+	}
+
+	reqID := ""
+	if opts != nil && pOpts.ksID != "" {
+		reqID = pOpts.ksID
+	}
+
```

Comment:
This code is redundant. Better to do like below:
```
func newWriter(kmsStore storage.Store, opts ...PrivateKeyOpts) *storeWriter {
	pOpts := &privateKeyOpts{}

	for _, opt := range opts {
		opt(pOpts)
	}

	return &storeWriter{
		storage:           kmsStore,
		requestedKeysetID: pOpts.ksID,
	}
}
```

Reply:
updated